packages/devtools/lib/src/inspector/inspector_controller.dart

```diff
@@ -734,4 +757,27 @@ class InspectorController implements InspectorServiceClient {
     }
     parent?.maybeUpdateValueUI(valueRef);
   }
+
+  Future<void> expandAllNodesInDetailsTree() async {
+    await details.recomputeTreeRoot(
+      inspectorTree.selection.diagnostic,
+      details.inspectorTree.selection.diagnostic,
+      false,
+      subtreeDepth: maxJsInt,
+    );
+  }
+
+  Future<void> resetDetailsTree() async {
+    assert(details.inspectorTree.selection?.diagnostic != null);
+    details.inspectorTree.setState(() {
```

Comment:
the expandPath method on inspectorTree is also calling setState. It could be worth it to add resetTree method in inspectorTree that does both the collapse and path expand. That way setState is only called once and you keep more of the tree display business logic in inspectorTree as a minor bonus given how complex this controller already is.